cisco/mlspp
# Update the build system

This PR is a pretty significant refactor of the tests and build system, based on lessons learned from the [SFrame implementation](https://github.com/cisco/sframe) setup.  There are several major changes here:

* Convert tests to doctest
* Remove the simulator app
* Simplify CMakeLists and Makefiles
* Enable clang-tidy more elegantly than before

Each of these should probably have been a PR of its own, but I was in a hurry.  When reviewing, it may be helpful to review individual commits rather than the whole diff.
